packages/block-library/src/cover-image/index.js

```diff
@@ -117,146 +127,171 @@ export const settings = {
+										onChange={ setDimRatio }
+										min={ 0 }
+										max={ 100 }
+										step={ 10 }
+									/>
+								</PanelColorSettings>
+							</PanelBody>
+						</InspectorControls>
+					) }
+				</Fragment>
+			);
+
+			if ( ! url ) {
+				const hasTitle = ! RichText.isEmpty( title );
+				const icon = hasTitle ? undefined : 'format-image';
+				const label = hasTitle ? (
+					<RichText
+						tagName="h2"
+						value={ title }
+						onChange={ ( value ) => setAttributes( { title: value } ) }
+						inlineToolbar
 					/>
-					<Toolbar>
-						<MediaUpload
+				) : __( 'Cover Image' );
+
+				return (
+					<Fragment>
+						{ controls }
+						<MediaPlaceholder
+							icon={ icon }
+							className={ className }
+							labels={ {
+								title: label,
+								name: __( 'an image' ),
+							} }
 							onSelect={ onSelectImage }
+							accept="image/*"
 							allowedTypes={ ALLOWED_MEDIA_TYPES }
-							value={ id }
-							render={ ( { open } ) => (
-								<IconButton
-									className="components-toolbar__control"
-									label={ __( 'Edit image' ) }
-									icon="edit"
-									onClick={ open }
-								/>
-							) }
+							notices={ noticeUI }
+							onError={ noticeOperations.createErrorNotice }
 						/>
-					</Toolbar>
-				</BlockControls>
-				{ !! url && (
-					<InspectorControls>
-						<PanelBody title={ __( 'Cover Image Settings' ) }>
-							<ToggleControl
-								label={ __( 'Fixed Background' ) }
-								checked={ !! hasParallax }
-								onChange={ toggleParallax }
-							/>
-							<RangeControl
-								label={ __( 'Background Opacity' ) }
-								value={ dimRatio }
-								onChange={ setDimRatio }
-								min={ 0 }
-								max={ 100 }
-								step={ 10 }
-							/>
-						</PanelBody>
-					</InspectorControls>
-				) }
-			</Fragment>
-		);
-
-		if ( ! url ) {
-			const hasTitle = ! RichText.isEmpty( title );
-			const icon = hasTitle ? undefined : 'format-image';
-			const label = hasTitle ? (
-				<RichText
-					tagName="h2"
-					value={ title }
-					onChange={ ( value ) => setAttributes( { title: value } ) }
-					inlineToolbar
-				/>
-			) : __( 'Cover Image' );
+					</Fragment>
+				);
+			}
 
 			return (
 				<Fragment>
 					{ controls }
-					<MediaPlaceholder
-						icon={ icon }
-						className={ className }
-						labels={ {
-							title: label,
-							name: __( 'an image' ),
-						} }
-						onSelect={ onSelectImage }
-						accept="image/*"
-						allowedTypes={ ALLOWED_MEDIA_TYPES }
-						notices={ noticeUI }
-						onError={ noticeOperations.createErrorNotice }
-					/>
+					<div
+						data-url={ url }
+						style={ style }
+						className={ classes }
+					>
+						{ ( ! RichText.isEmpty( title ) || isSelected ) && (
+							<RichText
+								tagName="p"
+								className="wp-block-cover-image-text"
+								placeholder={ __( 'Write title…' ) }
+								value={ title }
+								onChange={ ( value ) => setAttributes( { title: value } ) }
```

Comment:
This is actually an example of what I referred to above in my nitpick of the year actually: looks like this is the exact same function as another anonymous, inline one passed to a prop above. Might as well combine 'em.